lib/chef/platform/resource_priority_map.rb

```diff
@@ -0,0 +1,37 @@
+class Chef
+  class Platform
+    class ResourcePriorityMap
+      include Singleton
+
+      def initialize
+        load_default_map
+      end
+
+      def get_priority_array(node, resource_name)
+        priority_map.get(node, resource_name.to_sym)
+      end
+
+      def set_priority_array(resource_name, priority_array, *filter)
+        priority resource_name.to_sym, priority_array.to_a, *filter
+      end
+
+      def priority(*args)
+        priority_map.set(*args)
+      end
+
+      private
+
+      def load_default_map
+        require 'chef/resources'
+
+        # MacOSX
+        priority :package, Chef::Resource::HomebrewPackage, os: "darwin"
```

Comment:
The homebrew_package resource and macports_package resource both provide the package resource on mac, this line gives homebrew_package a higher priority and breaks the tie.  Then users can also override this priority and switch it back to the macports_package resource.